### PR TITLE
test: add test suite structure, parsing tests, automation, and stress

### PR DESCRIPTION
**Pull request by yashkrishan**
… fixes

- Add tests/ structure: unit/ (auth, parsing) and integration-tests/ (auth, conversations, parsing, stress)
- Move auth tests from app/modules/auth/tests/ to tests/unit/auth/ and tests/integration-tests/auth/
- Add parsing unit tests (schema, content_hash, encoding_detector, helper, validator, repo_name_normalizer)
- Add parsing integration tests (router, service with mocks; real_parse with Postgres+Neo4j)
- Add stress tests (concurrent stream timing, stress async endpoints) and relax timing assertion
- Add scripts/run_tests.py and run_tests.sh for one-command phased runs (unit → integration → real_parse → stress)
- Add docs/test-suite-and-parsing-plan.md with plan and Phase 4 checklist
- Update tests/conftest.py (fixtures, mock Redis/Celery, test_repo_path as git repo)
- Update pyproject.toml pytest markers (unit, integration, stress, real_parse, github_live)
- Fix Project status CheckConstraint to include created, processing, inferring (for real_parse tests)
- Add tests/README.md with run instructions and env vars (SKIP_REAL_PARSE, RUN_STRESS)

## Summary by CodeRabbit

* **New Features**
  * Projects gain an explicit "created" status and support intermediate parsing states.

* **Bug Fixes / Reliability**
  * Streaming endpoints now await background task startup and avoid blocking the event loop for improved responsiveness and concurrency.

* **Tests**
  * Test suite reorganized and expanded (unit, integration, stress, real-parse); added a centralized test runner and a tests README.

### CODE REVIEW

File: tests/integration-tests/conversations/test_concurrent_stream_timing.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+  thread and yields via a queue; the mock consume_stream is used in that thread.
+
+Before fix: sync wait_for_task_start blocked the event loop → N requests took ~ N * wait_secs.
+After fix: wait_for_task_start runs in a thread → N requests take ~ wait_secs (concurrent).
+
+Run all: pytest tests/integration-tests/conversations/test_concurrent_stream_timing.py -v
+
+Before/after timing comparison (use -s to see output):
+  pytest tests/integration-tests/conversations/test_concurrent_stream_timing.py -k before_vs_after -v -s
+"""
```
Comment (Dsantra92) [inline tests/integration-tests/conversations/test_concurrent_stream_timing.py:16]: Fix the docstring please

File: tests/README.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  - `SKIP_REAL_PARSE=1` — Skip real_parse phase (e.g. CI without Neo4j).
+  - `RUN_STRESS=1` — Include stress tests after the main phases.
+
+**Subset runs (same script):**
+
+```bash
+./scripts/run_tests.sh --unit-only
+./scripts/run_tests.sh --integration-only
+./scripts/run_tests.sh --real-parse-only
+./scripts/run_tests.sh --stress-only
```
Comment (Dsantra92) [inline tests/README.md:27]: pytest can already do this without .sh scripts. Do we really need new .sh files for compartmentalizing tests?

Comment (yashkrishan) [reply tests/README.md:27]: wanted to make it simpler

**Review by dhirenmathur — CHANGES_REQUESTED** [review]
I am still manually reviewing, but here are prelimnary AI results. 

(I am okay with sequential tests being a thing for now - i would rather have coverage first and then fix the framework. But take a look at the other things below definitely) 

```
**Findings**
- `P1` The new suite is not isolated at the DB layer. [`tests/conftest.py#L92`](/tmp/potpie-pr680/tests/conftest.py#L92) creates one shared test DB for the whole session, and [`tests/conftest.py#L133`](/tmp/potpie-pr680/tests/conftest.py#L133) only opens/closes sessions without rollback/truncation. The conversation tests already work around leaked state instead of asserting real emptiness in [`test_conversations_router.py#L753`](/tmp/potpie-pr680/tests/integration-tests/conversations/test_conversations_router.py#L753) and [`test_conversations_router.py#L769`](/tmp/potpie-pr680/tests/integration-tests/conversations/test_conversations_router.py#L769). That makes the suite order-dependent and weakens any “empty state” coverage.

- `P1` The concurrent-stream timing test can pass even when the claimed fix is absent. The test says `wait_for_task_start` is offloaded with `asyncio.to_thread` in [`test_concurrent_stream_timing.py#L4`](/tmp/potpie-pr680/tests/integration-tests/conversations/test_concurrent_stream_timing.py#L4), but the production path still calls `wait_for_task_start` synchronously in [`conversation_routing.py#L168`](/tmp/potpie-pr680/app/modules/conversations/utils/conversation_routing.py#L168). The monkeypatch in [`test_concurrent_stream_timing.py#L156`](/tmp/potpie-pr680/tests/integration-tests/conversations/test_concurrent_stream_timing.py#L156) targets an unused symbol, and the threshold in [`test_concurrent_stream_timing.py#L37`](/tmp/potpie-pr680/tests/integration-tests/conversations/test_concurrent_stream_timing.py#L37) is loose enough that serialized behavior can still stay green.

- `P2` The new harness still forces “pure unit” tests through the full app bootstrap. [`tests/conftest.py#L40`](/tmp/potpie-pr680/tests/conftest.py#L40) imports `app.main`, which imports [`database.py#L13`](/tmp/potpie-pr680/app/core/database.py#L13) and hard-fails without `POSTGRES_SERVER`, while [`main.py#L59`](/tmp/potpie-pr680/app/main.py#L59) exits unless `ENV=development`. That directly contradicts the README claim in [`tests/README.md#L88`](/tmp/potpie-pr680/tests/README.md#L88) that pure parsing unit tests run without Postgres, and it makes the framework much heavier than it needs to be.

- `P2` The auth replacement suite does not actually lock down the core multi-provider contract. [`test_auth_router.py#L42`](/tmp/potpie-pr680/tests/integration-tests/auth/test_auth_router.py#L42) accepts both `"new_user"` and `"success"` for the new-user flow, and [`test_auth_router.py#L60`](/tmp/potpie-pr680/tests/integration-tests/auth/test_auth_router.py#L60) accepts `"needs_linking"`, `"pending_link"`, or `"success"` for the existing-user/linking flow. Those tests never verify the DB side effects that matter for [`auth_router.py#L564`](/tmp/potpie-pr680/app/modules/auth/auth_router.py#L564) and [`auth_router.py#L618`](/tmp/potpie-pr680/app/modules/auth/auth_router.py#L618): provider creation, pending-link creation, or duplicate-account prevention.

- `P2` Several new edge-case tests are really smoke tests, not contract tests. Examples: [`test_parsing_router.py#L170`](/tmp/potpie-pr680/tests/integration-tests/parsing/test_parsing_router.py#L170) accepts `200/404/422` for the same path-like input, [`test_parsing_router.py#L201`](/tmp/potpie-pr680/tests/integration-tests/parsing/test_parsing_router.py#L201) accepts either same or different project IDs for double-submit, and [`test_conversations_router.py#L528`](/tmp/potpie-pr680/tests/integration-tests/conversations/test_conversations_router.py#L528) / [`test_conversations_router.py#L726`](/tmp/potpie-pr680/tests/integration-tests/conversations/test_conversations_router.py#L726) explicitly allow success on nonexistent conversations. Those won’t catch regressions in API behavior.

**Coverage Gaps**
- Auth: the PR removes the old `AuthService` unit coverage and nothing under `tests/` now replaces [`auth_service.py#L15`](/tmp/potpie-pr680/app/modules/auth/auth_service.py#L15), [`auth_service.py#L36`](/tmp/potpie-pr680/app/modules/auth/auth_service.py#L36), or [`auth_service.py#L61`](/tmp/potpie-pr680/app/modules/auth/auth_service.py#L61). There is no new coverage for `/login`, `/signup`, `create_custom_token`, development-mode auth fallback, personal-email blocking, unknown SSO provider, or provider-not-found cases for set-primary/unlink.

- Conversations: happy-path coverage is decent for create/message/regenerate CRUD, but there is no meaningful coverage for `resume_session`, non-streaming message flow, `background=False` regenerate flow, attachment upload cleanup on partial failure, real Redis `SessionService` parsing, or stop-generation persistence/Redis cleanup. The biggest module also is not marked `integration` in [`test_conversations_router.py#L11`](/tmp/potpie-pr680/tests/integration-tests/conversations/test_conversations_router.py#L11), so `pytest -m integration` would miss it.

- Parsing router/controller: the suite never exercises the most complex branches in [`parsing_controller.py#L60`](/tmp/potpie-pr680/app/modules/parsing/graph_construction/parsing_controller.py#L60), [`parsing_controller.py#L127`](/tmp/potpie-pr680/app/modules/parsing/graph_construction/parsing_controller.py#L127), or [`parsing_controller.py#L438`](/tmp/potpie-pr680/app/modules/parsing/graph_construction/parsing_controller.py#L438): path auto-detection, demo-repo duplication, and stuck-task auto-recovery. Those are exactly the kinds of flows that regress quietly.

- Parsing service/project lookup: mocked service tests cover exception plumbing, but not the commit-vs-branch lookup contract in [`projects_service.py#L241`](/tmp/potpie-pr680/app/modules/projects/projects_service.py#L241), READY short-circuit behavior, repo-manager worktree ensure, or GitHub-token fallback paths in [`parsing_service.py#L137`](/tmp/potpie-pr680/app/modules/parsing/graph_construction/parsing_service.py#L137).

- Real parse: [`test_real_parse.py#L28`](/tmp/potpie-pr680/tests/integration-tests/parsing/test_real_parse.py#L28) proves “status becomes READY/PARSED” and [`test_real_parse.py#L93`](/tmp/potpie-pr680/tests/integration-tests/parsing/test_real_parse.py#L93) proves “some nodes exist”, but it still doesn’t assert relationships, inference completion, repo scoping cleanup, or idempotent reparse behavior.

**Framework**
- `pytest` is still the right framework here. The inefficiency is in fixture architecture and test oracles, not in the choice of runner.
- The biggest efficiency wins are: split `conftest` by layer so pure units never import `app.main`; use per-test transaction rollback/truncation; stop reusing fixed IDs as shared state; keep timing/load checks out of normal pytest and move them to `pytest-benchmark`, Locust, or k6.
- For completeness rather than speed, the best add-ons are `Hypothesis`/`Schemathesis` for parser/router contracts and `testcontainers` for hermetic Postgres/Redis/Neo4j integration runs.
- I would not switch to `unittest`; I would keep `pytest` and make the harness lighter and stricter.

I validated the pure parsing-unit subset with `env ENV=development POSTGRES_SERVER=... uv run pytest ...`: 63 tests passed. The same subset failed without those env vars, which confirms the harness issue above. I did not run the DB-backed integration or real-parse suites end-to-end here.
```
